### DISCUSSION TITLE
Private networks exclusion for client

### DISCUSSION THREAD

**Discussion by lexandrop** [strongswan/strongswan#911] [Q&A] [answered]
Hi.
Currently my IPsec IKEv2 clients route all their traffic via IPsec.
I'm trying to exclude [private networks](https://datatracker.ietf.org/doc/html/rfc1918#section-3) from IPsec, so clients will use IPsec for everything except 192.168.0.0/16 etc.

Following [this advice](https://serverfault.com/a/304791/439749), I've managed to do it by specifying not
```
leftsubnet=0.0.0.0/0
```
, but
```
leftsubnet=0.0.0.0/5,8.0.0.0/7,11.0.0.0/8,12.0.0.0/6,16.0.0.0/4,32.0.0.0/3,64.0.0.0/2,128.0.0.0/3,160.0.0.0/5,168.0.0.0/6,172.0.0.0/12,172.32.0.0/11,172.64.0.0/10,172.128.0.0/9,173.0.0.0/8,174.0.0.0/7,176.0.0.0/4,192.0.0.0/9,192.128.0.0/11,192.160.0.0/13,192.169.0.0/16,192.170.0.0/15,192.172.0.0/14,192.176.0.0/12,192.192.0.0/10,193.0.0.0/8,194.0.0.0/7,196.0.0.0/6,200.0.0.0/5,208.0.0.0/4
```

But then I've found that strongSwan, like libreSwan, has "virtual_private" config directive ("virtual-private" in libreSwan's case). But I can't find any practical examples of it's usage so far.

Can I somehow manage to use this directive to exclude some networks/subnets from IPsec for clients?

**Comment by tobiasbrunner** [comment, marked as answer]
> But then I've found that strongSwan, like libreSwan, has "virtual_private" config directive ("virtual-private" in libreSwan's case). But I can't find any practical examples of it's usage so far.

Because, one, strongSwan doesn't support this option, and two, it doesn't do what you think it does.

> Can I somehow manage to use this directive to exclude some networks/subnets from IPsec for clients?

No, it's used to accept client traffic selectors from certain subnets if no [virtual IPs](https://wiki.strongswan.org/projects/strongswan/wiki/VirtualIp) are used (i.e. accept their private IP address as remote traffic selector). Since that causes all sorts of routing issues, virtual IPs are definitely recommended for clients behind NATs.


If you can't configure the clients to exclude private subnets themselves (e.g. via routing or bypass policies, for instance, via [bypass-lan plugin](https://wiki.strongswan.org/projects/strongswan/wiki/bypass-lan)), configuring a traffic selector that lists all public subnets is the only option (note that you are missing `224.0.0.0/3` in the above list). The only problem is that there are many clients that don't support this kind of split-tunneling (or split-tunneling at all).
